### PR TITLE
feat(ai-proxy): add option to return model name in response header

### PR DESCRIPTION
**Pull request by fffonion**
### Checklist

- [x] The Pull Request has tests
- [x] A changelog file has been created under `changelog/unreleased/kong` or `skip-changelog` label added on PR if changelog is unnecessary. [README.md](https://github.com/Kong/gateway-changelog/blob/main/README.md)
- [ ] There is a user-facing docs PR against https://github.com/Kong/docs.konghq.com - PUT DOCS PR HERE

### Issue reference

Fix #_[issue number]_

### CODE REVIEW

**Review by oowl — APPROVED** [review]

**Review by liverpool8056 — CHANGES_REQUESTED** [review]

File: kong/llm/proxy/handler.lua [hunk 1]
```diff
@@ -335,6 +335,11 @@ function _M:header_filter(conf)
     kong.response.clear_header(v)
   end
 
+  if ngx.var.http_kong_debug or conf.model_name_header then
+    local name = conf.model.provider .. "/" .. (kong.ctx.plugin.llm_model_requested or conf.model.name)
```
[the remark this answers is not in the document]
Comment (fffonion) [reply kong/llm/proxy/handler.lua:339]: yes, `model.name` can be overriden by request, but not `model.provider`.

File: spec/02-integration/09-hybrid_mode/09-config-compat_spec.lua [hunk 2]
```diff
@@ -482,7 +482,7 @@ describe("CP/DP config compat transformations #" .. strategy, function()
     end)
 
     describe("ai plugins supported providers", function()
-      it("[ai-proxy] tries to use unsupported providers on older Kong versions", function()
+      it("[ai-proxy] tries to use unsupported providers on older Kong versions #ttt", function()
```
[the remark this answers is not in the document]
Comment (fffonion) [reply spec/02-integration/09-hybrid_mode/09-config-compat_spec.lua]: yes, my fault

File: kong/clustering/compat/removed_fields.lua [hunk 3]
```diff
@@ -175,6 +175,7 @@ return {
       "model.options.bedrock",
       "auth.aws_access_key_id",
       "auth.aws_secret_access_key",
+      "model_header_name",
```
Comment (StarlightIbuki) [inline kong/clustering/compat/removed_fields.lua]: `model_name_header` or `model_header_name`?

**Review by StarlightIbuki — CHANGES_REQUESTED** [review]

**Review by StarlightIbuki — APPROVED** [review]

**Review by liverpool8056 — APPROVED** [review]

**Comment by fffonion** [comment]
will do cherrypick as part of https://github.com/Kong/kong-ee/pull/9912

**Pull request merged**